ComponentKit/Utilities/CKComponentAction.h

```diff
@@ -11,44 +11,180 @@
 #import <UIKit/UIKit.h>
 
 #import <ComponentKit/CKComponentViewAttribute.h>
+#import <ComponentKit/CKComponentActionInternal.h>
 
 @class CKComponent;
 
 /**
- A component action is simply a selector. The selector may optionally take one argument: the sending component.
+ CKTypedComponentAction is a struct that represents a method invocation that can be passed to a child component to
+ trigger a method invocation on a target.
+ 
+ We allow a typed specification of the parameters that will be provided as arguments to the component action at runtime
+ through the variadic templated arguments. You may specify an arbitrary number of arguments to your component action,
+ and you may use either object, or primitive arguments. Only trivially-constructible arguments or pointers can be used
```

Comment:
I wonder whether people understand "trivially-constructible". It's a C++ oriented (or maybe even specific) term. I didn't know about this, but it made sense once I read the docs.